R/bar-graph-all-years.R

```diff
@@ -0,0 +1,117 @@
+  for (cchs_str in separated_vars) {
+    cchs_year <- eval(parse(text=paste(cchs_str)))
+    rwt_data <- rec_with_table(cchs_year,
+                               database_name = cchs_str,
+                               var_name_str,
+                               log = TRUE)
+    # Define the year as only the year-number in the database name string
+    if (nchar(cchs_str)>= 15)
+    {
+      yr_str <- substring(cchs_str, 5, 13)
+      yr_str <- gsub("_","-", yr_str)
+    } else {
+      yr_str <- substring(cchs_str, 5, 8)
+    }
+    if (k == 1) {
+      cchs_data <- as_label(rwt_data)
+      cchs_data$year <- yr_str
+    } else {
+      cchs_data <- bind_rows(cchs_data,as_label(rwt_data))
+      cchs_data$year[is.na(cchs_data$year)] <- yr_str
+    }
+    k <- k+1
+  }
+
+  if (ncol(cchs_data) > 1) {
+    # Generate stacked barplot for categorical variables
+    if (variables$variableType[i] == "Categorical") {
+      # Create table with number of each categorical response for the plot
+      agg <- count(cchs_data,eval(parse(text=paste(var_name_str))),year)
+      names(agg)[1] <- var_name_str
+      plot <- ggplot(agg, aes(x = year,
+                              y = n,
+                              fill = eval(parse(text=paste(var_name_str))))) + 
+                geom_bar(position="stack", stat="identity") +
+                labs(
+                  fill = var_name_str,
+                  x = "Year",
+                  y = var_name_str
+      )
+      # Transform data with tidyr::spread() for the variable name
+      agg_chi_sq <- spread(agg,var_name_str,n)
+      rownames(agg_chi_sq) <- agg_chi_sq[,1]
+      agg_chi_sq <- agg_chi_sq[,-1]
+      agg_chi_sq[is.na(agg_chi_sq)] <- 0
+      # Run chi squared and separate significance based on p value
+      chisq <- chisq.test(agg_chi_sq)
+      p <- chisq$p.value
+    } else {
+      # Create bar graphs for continuous variables for the means
+      if (variables$variableType[i] == "Continuous") {
+        plot <- ggplot(cchs_data, aes(x = year, 
+                                      y = eval(parse(text=paste(var_name_str))), 
+                                      group = year, 
+                                      fill = year)) +
+          geom_boxplot(alpha = .7) +
+          geom_jitter(width = .05, alpha = .4) +
+          guides(fill = "none") +
+          theme_bw() +
+          labs(
+            x = "Year",
+            y = var_name_str
+          )
+        # Run one-way ANOVA to determine significance for continuous variables
+        res.aov <- aov(eval(parse(text=paste(var_name_str))) ~ year, data = cchs_data)
+        p <- summary(res.aov)[[1]][["Pr(>F)"]]
+      } else {
+        warning("Variable type is neither categorical nor continuous.")
+      }
+    }
+    # Save significant and ns plots in different folders
+    if (p < 0.05) {
+      img_label <- paste("man/figures/sig_plots/", var_name_str, ".png", 
```

Comment:
The location to save them should also be an argument for the overarching function